### ISSUE TITLE
False positive for TS2448 "use before declaration" in 3.8.3 --target esnext

### ISSUE THREAD

**Issue by Jessidhia** [microsoft/TypeScript#37287]
**TypeScript Version:**  3.8.3

**Search Terms:**
TS2448, use before declaration

**Code**

```ts
const a = () => b()
const b = () => null
a()
```

**Expected behavior:**
No errors

**Actual behavior:**
> Block-scoped variable 'b' used before its declaration.

**Playground Link:** [Playground Link](http://www.typescriptlang.org/play/index.html?useDefineForClassFields=true&noUnusedLocals=true&noUnusedParameters=true&target=99&jsx=1#code/MYewdgzgLgBAhjAvDAFASiQPhgI3QKFElhyVQ0WzAFcAbW-OdIA)

---

This error seems to _only_ happen, at least in the playground, when the `target` is set to `ESNext`. If it's set to, say, `ES2020`, the error goes away... in `tsservice`; it still happens in `tsc`.

This seems to have been introduced between `3.8.0-beta` (builds) and `3.8.1-rc` (errors).

**Comment by RyanCavanaugh** [comment]
@Jessidhia do you need a patch build for this?

**Comment by Jessidhia** [comment]
Probably not, I can just move to the nightly or 3.9.0 when it's fixed 🤔
